src/parcsr_ls/dsuperlu.c

```diff
@@ -174,12 +222,40 @@ hypre_SLUDistSolve(void            *solver,
       x_data = hypre_VectorData(hypre_ParVectorLocalVector(x));
    }
 
-   pdgssvx(&(dslu_data->dslu_options), &(dslu_data->A_dslu),
-           &(dslu_data->dslu_ScalePermstruct), x_data, size, nrhs,
-           &(dslu_data->dslu_data_grid), &(dslu_data->dslu_data_LU),
-           &(dslu_data->dslu_solve), dslu_data->berr, &(dslu_data->dslu_data_stat), &info);
+   /* SuperLU uses sbl to denote its floating point type. Hence, the conversion/checks below: */
+   if (sizeof(hypre_double) != sizeof(HYPRE_Complex))
+   {
+      slu_data = hypre_CTAlloc(hypre_double, size, HYPRE_MEMORY_HOST);
+      for (i = 0; i < size; i++)
+      {
+         slu_data[i] = (hypre_double) x_data[i];
+      }
+   }
+   else
+   {
+      slu_data = (hypre_double*) x_data;
+   }
+
+   pdgssvx(&(dslu_data->dslu_options),
+           &(dslu_data->A_dslu),
+           &(dslu_data->dslu_ScalePermstruct),
+           slu_data,
+           (int_t) size,
+           (int_t) nrhs,
+           &(dslu_data->dslu_data_grid),
+           &(dslu_data->dslu_data_LU),
+           &(dslu_data->dslu_solve),
+           dslu_data->berr,
+           &(dslu_data->dslu_data_stat),
+           &info);
+
+   /* Free memory */
+   if ((void*) slu_data != (void*) x_data)
+   {
+      hypre_TFree(slu_data, HYPRE_MEMORY_HOST);
```

Comment:
I don't see the free of `slu_data` in `SLUDistSetup` above, but I'm not sure I understand how memory is handled by SuperLU.  Just thought I'd mention it.

Reply:
Hi Rob, in the setup routine, SuperLU takes ownership of the pointer, which is destroyed in the call `Destroy_CompRowLoc_Matrix_dist`  :)